doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -115,31 +112,53 @@ Currently 7 transceiver events are defined as below.
     status='5' High Temperature,
     status='6' Bad cable.
 
-However, multiple errors could exist at the same time. For example, a module can be unsupported cable and high temperature. The new transceiver event will be described in a bitmap. New transceiver definition below.
+However, multiple errors could exist at the same time. For example, a module can be unsupported cable and high temperature. The new transceiver event will be described in a bitmap as below:
 
     bit 32  : 0=SFP removed, 1=SFP inserted,
-    bit 31  : 0=OK, 1=I2C bus stuck,
-    bit 30  : 0=OK, 1=Bad eeprom,
-    bit 29  : 0=OK, 1=Unsupported cable,
-    bit 28  : 0=OK, 1=High Temperature,
-    bit 27  : 0=OK, 1=Bad cable.
-    bit 1~26: reserved. Must be 0.
+    bit 31  : 0=OK, 1=An error that blocks eeprom from being read,
+    bit 30  : 0=OK, 1=I2C bus stuck,
+    bit 29  : 0=OK, 1=Bad eeprom,
+    bit 28  : 0=OK, 1=Unsupported cable,
+    bit 27  : 0=OK, 1=High Temperature,
+    bit 26  : 0=OK, 1=Bad cable.
+    bit 1~25: reserved. Must be 0.
+
+    Define bit 32 as the least significant bit and bit 1 as the most significant bit.
+
+The bit 32 represents whether the SFP module is inserted. Any error bit must be set along with this bit.
+
+The bit 31 represents whether the error indicated by platform API will block the SFP module from being read or not. This bit must be set along with other error bits.
+
+Vendor can extend this bitmap with more errors. However, some errors can be vendor specific, which means they won't occur on other vendors' platforms. The bitmap can grow rapidly and eventually run out of bits if all vendors insert vendor specific error codes to the bitmap.
 
-Vendor can extend this bitmap with more errors. Xcvrd need parse the bitmap and set transceiver status table in DB accordingly. For example, if the transceiver event bit map is 0x7, the status field value should be "1" and the error field value should be "I2C bus stuck | Bad eeprom".
+This can be resolved by dividing the bitmap into two parts: one for generic errors and the other for vendor specific errors. The bits 17 ~ 30 represent the generic errors and bits 1 ~ 16 represent the vendor specific errors.

```

Comment:
Here you state that bits 1~16 represent vendor-specific bits, but above you state `bit 1~25: reserved. Must be 0.` This seems contradictory.